src/test/regress/expected/failure_ddl.out

```diff
@@ -0,0 +1,1228 @@
+-------------------------------
+                             0
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,2)
+ (localhost,57637,100802,t,2)
+ (localhost,57640,100801,t,2)
+ (localhost,57640,100803,t,2)
+(4 rows)
+
+-- kill as soon as the coordinator sends COMMIT
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- some of the placements would be missing the new column
+-- since we've not commited the prepared transactions
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     3
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,2)
+ (localhost,57640,100803,t,2)
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is committed
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             2
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- finally, test failing on ROLLBACK with 2CPC
+-- fail just after the coordinator sends the ROLLBACK
+-- so the command can be rollbacked
+SELECT citus.mitmproxy('conn.onQuery(query="ROLLBACK").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+BEGIN;
+ALTER TABLE test_table DROP COLUMN new_column;
+ROLLBACK;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- ROLLBACK should have failed on the distributed table and the placements
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     3
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- but now kill just after the worker sends response to 
+-- ROLLBACK command, so we'll have lots of warnings but the command
+-- should have been rollbacked both on the distributed table and the placements
```

Comment:
You probably don't need both of these `ROLLBACK` tests. If we're going to suppress WARNINGs then their behavior is identical. In one of them we explicitly send a ROLLBACK, in the other we just drop the connection and rely on workers to ROLLBACK themselves when the backend is killed.

Reply:
Okay, makes sense